### PR TITLE
Core: Ensure that `context.args` is always set

### PR DESCRIPTION
**Pull request by tmeasday**
Issue: #16743 

## What I did

Ensure we don't filter all args down to nothing.

## How to test

- [x] Is this testable with Jest or Chromatic screenshots?
Yes see tests

### CODE REVIEW

File: lib/store/src/prepareStory.test.ts [hunk 1]
```diff
@@ -527,6 +527,49 @@ describe('prepareStory', () => {
...
+        { id, title },
+        { render: renderMock }
+      );
+
+      firstStory.unboundStoryFn({
+        args: firstStory.initialArgs,
+        hooks: new HooksContext(),
+        ...firstStory,
+      } as any);
+      expect(renderMock).toHaveBeenCalledWith({}, expect.objectContaining({ argsByTarget: {} }));
```
Comment (shilman) [inline lib/store/src/prepareStory.test.ts:571]: @tmeasday shouldn't this be asserting something about `args`? and same for above?

Comment (tmeasday) [reply lib/store/src/prepareStory.test.ts:571]: Args is the first argument (`{}`)

Comment (shilman) [reply lib/store/src/prepareStory.test.ts:571]: aha! 💡

**Review by shilman — APPROVED** [review]
LGTM!

**Comment by borodovisin** [comment]
Hi @shilman, I have a question related to code override to the Story.parameters.docs.container function.
I'm upgrading to version 6.4.9 and now the `DocsContextProps` type has the `args` key deprecated. That means in the further releases that key won't exist anymore?

I need access to the args set by users on the Control table...

**Pull request merged**